doc/syntax/comments.rdoc

```diff
@@ -35,3 +35,188 @@ syntax error:
     Will not work
     =end
   end
+
+= Magic Comments
+
+While comments are typically ignored by Ruby, special "magic comments" contain
+directives that affect how the code is interpreted.
+
+Top-level magic comments must start on the first line, or on the second line if
+the first line looks like `#! shebang line`.
+
+NOTE: Magic comments affect only the file in which they appear;
+other files are unaffected.
+
+== Alternative syntax
+
+Magic comment may consist of a single directive multiple (as in the examples above).
+Alternatively, multiple directives may appear on the same line if separated by ";"
+and wrapped between "-*-" (See Emacs' file variables: https://www.gnu.org/software/emacs/manual/html_node/emacs/Specifying-File-Variables.html)
+
+  # -*- emacs-compatible; encoding: big5; frozen_string_literal: true -*-
+
+  p 'hello'.frozen? # => true
+  p 'hello'.encoding # => true
+
+== <code>encoding</code> Directive
+
+Indicates which string encoding should be used for string literals:
+
+  # encoding: big5
+
+  ''.encoding # => #<Encoding:Big5>
+
+Default encoding is UTF-8.
+
+It must appear in the first comment section of a file
+
+The word "coding" may be used instead of "encoding".
+
+== <code>frozen_string_literal</code> Directive
+
+When appears in the top section of a file, indicates that string literals should be allocated once at parse time and frozen.
+
+  # frozen_string_literal: true
+
+  3.times do
+    p 'hello'.object_id # => prints same number
+  end
+  p 'world'.frozen? # => true
+
+The default is false. Without the comment, or with
+<code># frozen_string_literal: false</code>,
+the example above would print 3 different numbers and "false".
+
+Starting in Ruby 3.0, string literals that are dynamic are not frozen nor reused:
+
+  # frozen_string_literal: true
+
+  p "Addition: #{2 + 2}".frozen? # => false
+
+It must appear in the first comment section of a file
```

Comment:
Probably relationships with command-line option `--enable=frozen-string-literal` should be mentioned.

Reply:
Definitely, I don't know about them 😆. Too tired to look right now... tomorrow hopefully

Reply:
Mentionned, thanks!